src/main/scala/io/iohk/ethereum/vm/Memory.scala

```diff
@@ -0,0 +1,89 @@
+package io.iohk.ethereum.vm
+
+import akka.util.ByteString
+import cats.syntax.either._
```

Comment:
Please take a note of my comment before approving this PR. `cats` is not needed here, and we're going to remove it in the main PR.